### PR TITLE
qa: setting the config options value back to default at the end of th…

### PR DESCRIPTION
**Pull request by neesingh-rh**
…e tests

In order to avoid test failures due to setting the config 
option values different from default in a particular test
because the configs are retained across tests.

signed-off-by: Neeraj Pratap Singh <neesingh@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: qa/tasks/cephfs/test_volumes.py
Comment (rishabh-d-dave) [inline qa/tasks/cephfs/test_volumes.py:5862]: Comment `# set the config options back to default` is redundant, method name is sufficient.

**Review by rishabh-d-dave — CHANGES_REQUESTED** [review]

**Review by vshankar — CHANGES_REQUESTED** [review]
@neesingh-rh This requires a tracker ticket as it needs to be backported, yes?

Also, please add `Introduced-by: sha` in the commit message mentioning the SHA of the commit that introduced the issue.

File: qa/tasks/cephfs/test_volumes.py
Comment (batrick) [inline qa/tasks/cephfs/test_volumes.py:314]: This should not be necessary. See `CephTestCase.tearDown` and `.config_clear`.

**Review by batrick — CHANGES_REQUESTED** [review]

File: qa/tasks/cephfs/test_volumes.py
Comment (neesingh-rh) [reply qa/tasks/cephfs/test_volumes.py:314]: We encountered this issue while testing https://github.com/ceph/ceph/pull/52670. We didn't get the config options cleared within a test class. Here's the link: https://github.com/ceph/ceph/pull/52670#discussion_r1484152748

**Comment by neesingh-rh** [comment]
> @neesingh-rh This requires a tracker ticket as it needs to be backported, yes?
Should we backport it as it is in a way just a cleanup.
> Also, please add `Introduced-by: sha` in the commit message mentioning the SHA of the commit that introduced the issue.
And it is not introduced by any commit, we just encountered the issue during our testing. If you say I will link that discussion link.

File: qa/tasks/cephfs/test_volumes.py
Comment (joscollin) [reply qa/tasks/cephfs/test_volumes.py:314]: Sadly, these are maintained inside mgr/volumes. Is there any way to move it to src/common/options yaml ?

Comment (vshankar) [reply qa/tasks/cephfs/test_volumes.py:314]: @neesingh-rh Is there a link to a failed run that prompted this change?

Comment (neesingh-rh) [reply qa/tasks/cephfs/test_volumes.py:314]: Here's the failure link: https://github.com/ceph/ceph/pull/52670#pullrequestreview-1841436109

Comment (vshankar) [reply qa/tasks/cephfs/test_volumes.py:314]: @neesingh-rh That failure was reported on Jan 24 and the PR was merged on Feb 20 - are you saying that the failures wasn't RCA'd and the change was approved and merged (the tests passed after you pushed in some fixes)?

Comment (neesingh-rh) [reply qa/tasks/cephfs/test_volumes.py:314]: The issue we faced for our particular tests got fixed by reverting the configs to the default at the end of the tests which were part of  this particular https://github.com/ceph/ceph/pull/52670 PR. But https://github.com/ceph/ceph/pull/57627 fix adds a function which disables all config options at the end of every tests from now-onwards, so that we don't face this issue again in future.
See this comment where it is advised to revert the configs at the end of every tests: https://github.com/ceph/ceph/pull/52670#discussion_r1484152748

Comment (vshankar) [reply qa/tasks/cephfs/test_volumes.py:314]: In that entire run, there is just one invocation of `config rm` for `snapshot_clone_no_wait`.

```
zgrep "config rm" teuthology.log.gz  | grep snapshot_clone_no_wait
2024-01-22T08:27:52.269 DEBUG:teuthology.orchestra.run.smithi031:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph config rm mgr mgr/volumes/snapshot_clone_no_wait
2024-01-22T08:27:52.620 INFO:teuthology.orchestra.run.smithi031.stderr:2024-01-22T08:27:52.620+0000 7f4fef20f700  1 -- 172.21.15.31:0/3759669512 --> [v2:172.21.15.31:3300/0,v1:172.21.15.31:6789/0] -- mon_command({"prefix": "config rm", "who": "mgr", "name": "mgr/volumes/snapshot_clone_no_wait"} v 0) v1 -- 0x7f4fe8068230 con 0x7f4fe80d0920
2024-01-22T08:27:52.623 INFO:teuthology.orchestra.run.smithi031.stderr:2024-01-22T08:27:52.624+0000 7f4fe57fa700  1 -- 172.21.15.31:0/3759669512 <== mon.0 v2:172.21.15.31:3300/0 7 ==== mon_command_ack([{"prefix": "config rm", "who": "mgr", "name": "mgr/volumes/snapshot_clone_no_wait"}]=0  v73) v1 ==== 117+0+0 (secure 0 0 0) 0x7f4fe0059c50 con 0x7f4fe80d0920
```

@neesingh-rh Suggest debugging into why that happened - we should have seen that config removed for tests that did a `config set` for the config (which seems to be a handful of tests).

**Review by leonid-s-usov — CHANGES_REQUESTED** [review]
I agree with Patrick. There should not be a reason to update multiple tests with a code line that must be part of the teardown.

File: qa/tasks/cephfs/test_volumes.py
Comment (neesingh-rh) [reply qa/tasks/cephfs/test_volumes.py:314]: > In that entire run, there is just one invocation of `config rm` for `snapshot_clone_no_wait`.
> 
> ```
> zgrep "config rm" teuthology.log.gz  | grep snapshot_clone_no_wait
> 2024-01-22T08:27:52.269 DEBUG:teuthology.orchestra.run.smithi031:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph config rm mgr mgr/volumes/snapshot_clone_no_wait
> 2024-01-22T08:27:52.620 INFO:teuthology.orchestra.run.smithi031.stderr:2024-01-22T08:27:52.620+0000 7f4fef20f700  1 -- 172.21.15.31:0/3759669512 --> [v2:172.21.15.31:3300/0,v1:172.21.15.31:6789/0] -- mon_command({"prefix": "config rm", "who": "mgr", "name": "mgr/volumes/snapshot_clone_no_wait"} v 0) v1 -- 0x7f4fe8068230 con 0x7f4fe80d0920
> 2024-01-22T08:27:52.623 INFO:teuthology.orchestra.run.smithi031.stderr:2024-01-22T08:27:52.624+0000 7f4fe57fa700  1 -- 172.21.15.31:0/3759669512 <== mon.0 v2:172.21.15.31:3300/0 7 ==== mon_command_ack([{"prefix": "config rm", "who": "mgr", "name": "mgr/volumes/snapshot_clone_no_wait"}]=0  v73) v1 ==== 117+0+0 (secure 0 0 0) 0x7f4fe0059c50 con 0x7f4fe80d0920
> ```
> 
> @neesingh-rh Suggest debugging into why that happened - we should have seen that config removed for tests that did a `config set` for the config (which seems to be a handful of tests).

I am on it today.

Comment (neesingh-rh) [reply qa/tasks/cephfs/test_volumes.py:314]: > In that entire run, there is just one invocation of `config rm` for `snapshot_clone_no_wait`.
> 
> ```
> zgrep "config rm" teuthology.log.gz  | grep snapshot_clone_no_wait
> 2024-01-22T08:27:52.269 DEBUG:teuthology.orchestra.run.smithi031:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph config rm mgr mgr/volumes/snapshot_clone_no_wait
> 2024-01-22T08:27:52.620 INFO:teuthology.orchestra.run.smithi031.stderr:2024-01-22T08:27:52.620+0000 7f4fef20f700  1 -- 172.21.15.31:0/3759669512 --> [v2:172.21.15.31:3300/0,v1:172.21.15.31:6789/0] -- mon_command({"prefix": "config rm", "who": "mgr", "name": "mgr/volumes/snapshot_clone_no_wait"} v 0) v1 -- 0x7f4fe8068230 con 0x7f4fe80d0920
> 2024-01-22T08:27:52.623 INFO:teuthology.orchestra.run.smithi031.stderr:2024-01-22T08:27:52.624+0000 7f4fe57fa700  1 -- 172.21.15.31:0/3759669512 <== mon.0 v2:172.21.15.31:3300/0 7 ==== mon_command_ack([{"prefix": "config rm", "who": "mgr", "name": "mgr/volumes/snapshot_clone_no_wait"}]=0  v73) v1 ==== 117+0+0 (secure 0 0 0) 0x7f4fe0059c50 con 0x7f4fe80d0920
> ```
> 
> @neesingh-rh Suggest debugging into why that happened - we should have seen that config removed for tests that did a `config set` for the config (which seems to be a handful of tests).

After spending time with the changes in PR and logs, I found that one invocation of `config rm` for `snapshot_clone_no_wait` is expected. As some additions of `config_set` for `snapshot_clone_no_wait` were done after that particular run and to be precise after this PR got merged. See this: https://github.com/ceph/ceph/pull/55660 .
And `test_subvolume_snapshot_clone_when_no_wait_enabled` fails and only one test i.e. `test_subvolume_snapshot_clone_ cancel_pending` containing `config_set` for `snapshot_clone_no_wait` succeeds for which we saw `config rm` .

Comment (neesingh-rh) [reply qa/tasks/cephfs/test_volumes.py:314]: I checked working of `.config_clear` through vstart_runner as well as teuthology. It seems it works perfectly and c`config rm` happens after each and every particular tests.

**Comment by vshankar** [comment]
@neesingh-rh I don't understand the scope of this fix now given that .config_clear will reset any set configs. Have you checked one of the latest runs to validate this?

**Comment by neesingh-rh** [comment]
> @neesingh-rh I don't understand the scope of this fix now given that .config_clear will reset any set configs. Have you checked one of the latest runs to validate this?

Yup, It seems we should close this PR , as `.config_clear` is perfectly working and doing the deletion.

**Comment by neesingh-rh** [comment]
Closing this PR as per https://github.com/ceph/ceph/pull/57627#issuecomment-2201886182 and https://github.com/ceph/ceph/pull/57627#issuecomment-2199938211

**Pull request closed without merging**
